playground/tsconfig-json/src/decorator.ts

```diff
@@ -4,8 +4,7 @@ function first() {
 
 export class Foo {
   @first()
-  // @ts-expect-error we intentionally not enable `experimentalDecorators` to test esbuild compat
```

Comment:
experimentalDecorators is already enabled by https://github.com/vitejs/vite/pull/14804/files#diff-4a47aee52b9ebc021872b1ab146d0d7bfd3706b2c197a1de75ae245f84c143afR18

It seems `@first` needs to be `@first()`. There was a parameter count mismatch error.

Reply:
I remember `@first` should also work too, but I'm not sure if it's because esbuild doesn't implement it or something. TypeScript and the decorators spec documents that it would work:

- https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators
- https://github.com/tc39/proposal-decorators/blob/master/EXTENSIONS.md#parameter-decorators-and-annotations

Reply:
The examples in the TypeScript doc have these types:
```ts
function required(target: Object, propertyKey: string | symbol, parameterIndex: number): void
function validate(target: any, propertyName: string, descriptor: TypedPropertyDescriptor<Function>): void
```
The example in the decorators spec has this types:
```ts
function dec(_, context): Function {}
```

So I guess the type of `first` (`() => (...args: any[]) => void`) is allowed for the stage-3 decorators but not for TypeScript experimental decorators.

Reply:
As long as the test would fail like before (if we didn't implement it right), I'm ok with this change then 👍